### PR TITLE
Speedup CI tests

### PR DESCRIPTION
**Pull request by ghost**
Will try to improve the CI performance.

### CODE REVIEW

**Comment by grekiki** [comment]
Still takes a while if test.yaml is changed, but assuming the cache is found it should take around 10-11 minutes instead of 13-14

**Comment by grekiki** [comment]
No idea why test_startup has inconsistent behavior. Fingerprinting just randomly fails sometimes

**Comment by grekiki** [comment]
I think this should be good.

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -191,28 +191,284 @@ jobs:
...
                     $UNIT_TEST selfdrive/thermald && \
-                    $UNIT_TEST tools/lib/tests"
+                    $UNIT_TEST tools/lib/tests && \
+                    $UNIT_TEST selfdrive/controls"
+                    
       - name: Upload coverage to Codecov
         run: |
           docker commit tmppilot tmppilotci
           $CI_RUN "cd /tmp/openpilot && bash <(curl -s https://codecov.io/bash) -v -F unit_tests"
+  unit_tests1:
```
Comment (adeebshihadeh) [inline .github/workflows/test.yaml:223]: why are there multiple copies of this job?

Comment (grekiki) [reply .github/workflows/test.yaml:223]: Reliability testing, will remove before merge

Comment (grekiki) [reply .github/workflows/test.yaml]: removed now

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -191,28 +193,62 @@ jobs:
...
-                    $UNIT_TEST tools/lib/tests"
+                    $UNIT_TEST tools/lib/tests && \
+                    $UNIT_TEST selfdrive/controls"
+                    
       - name: Upload coverage to Codecov
         run: |
           docker commit tmppilot tmppilotci
           $CI_RUN "cd /tmp/openpilot && bash <(curl -s https://codecov.io/bash) -v -F unit_tests"
+                                                                                
+  loggerd_tests:
```
Comment (adeebshihadeh) [inline .github/workflows/test.yaml:226]: the uploader tests was just sleeping for a ridiculous amount of time. fixing in another PR

**Comment by adeebshihadeh** [comment]
Closing this, fixed some stuff up in #19621

**Pull request closed without merging**